dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -401,6 +407,30 @@ private void updateTags(List<Tag> newTags, @NotNull Tool tool, SourceCodeRepoInt
         toolDAO.create(tool);
     }
 
+    private void updateImageInformation(Tool tool, Tag newTag, Tag oldTag) {
```

Comment:
Is the complexity of the entire function just to save a possibly unneeded DB delete?

Reply:
Yeah, I guess I only wanted to delete and update the image in db if needed. But I can change it to update the tag's image every time if that's preferable.

```
if (oldTag.getImages().isEmpty()) {
            oldTag.getImages().addAll(updateTagWithImageInformation(tool, newTag).getImages());

        } else {
            oldTag.getImages().removeAll(oldTag.getImages());
            oldTag.getImages().addAll(newTag.getImages());
```

Reply:
It's ok either way, I was just wondering